tracking/src/main/java/org/hps/recon/tracking/kalman/KalmanPatRecDriver.java

```diff
@@ -91,130 +94,229 @@ public class KalmanPatRecDriver extends Driver {
+        }
+    }
+
+    /**
+     * Option to treat the B field as uniform. This normally is used only for
+     * development and debugging work.
+     *
+     * @param input Set true to assume a uniform B field.
+     */
+    public void setUniformB(boolean uniformB) {
+        this.uniformB = uniformB;
+        if (uniformB) {
+            System.out.println("KalmanPatRecDriver: a uniform field will be used in track fitting: %b");
+        }
     }
 
-    public void setUniformB(boolean input) {
-        uniformB = input;
-        logger.config("KalmanPatRecDriver: the B field will be assumed uniform.\n");
+    public void setELoss(boolean eLoss) {
+        this.eLoss = eLoss;
+        if (eLoss) {
+            System.out.println("KalmanPatRecDriver: energy loss will be included in fits.");
+        }
     }
 
     public void setMaterialManager(MaterialSupervisor mm) {
         _materialManager = mm;
     }
 
+    /**
+     * Determine whether residuals will be added to the event output
+     *
+     * @param input set true to force residuals to be output with the event
+     */
+    public void setAddResiduals(boolean addResiduals) {
+        this.addResiduals = addResiduals;
+        System.out.format("KalmanPatRecDriver: residuals will be added to the event output: %b\n", addResiduals);
+    }
+
+    /**
+     * Do-nothing method. See instead setOutputFullTrackCollectionName.
+     *
+     * @param input ignored---the call will do nothing.
+     */
     public void setTrackCollectionName(String input) {
     }
-    
+
+    /**
+     * Set the maximum number of SiClusters in one event allowed for KF pattern
+     * recognition (protection against monster events)
+     *
+     * @param input Maximum number of hits to use
+     */
     public void setSiHitsLimit(int input) {
         siHitsLimit = input;
-    }            
-
-    public void setAddResiduals(boolean input) {
-        addResiduals = input;
     }
-    
+
+    /**
+     * Set up the geometry and parameters for the Kalman-filter track finding
+     * and fitting.
+     */
     @Override
     public void detectorChanged(Detector det) {
         logger = Logger.getLogger(KalmanPatRecDriver.class.getName());
         if (logLevel != null) {
             logger.setLevel(logLevel);
             //LogManager.getLogManager().getLogger(Logger.GLOBAL_LOGGER_NAME).setLevel(logLevel);
         }
-        verbose = (logger.getLevel()==Level.FINE);
+        verbose = (logger.getLevel() == Level.FINE);
         System.out.format("KalmanPatRecDriver: entering detectorChanged, logger level = %s\n", logger.getLevel().getName());
         executionTime = 0.;
         interfaceTime = 0.;
         plottingTime = 0.;
         maxTime = 0.;
-        
+
         _materialManager = new MaterialSupervisor();
         _materialManager.buildModel(det);
 
         fm = det.getFieldMap();
-        /*
-        System.out.format("B field map vs y:\n");
+
```

Comment:
It appears to me lines 204 through 226 are for printing the b-field for debugging purposes. It was commented out before, can we do that again, or add a debug parameter to make it configurable to dump this info and make the default to not print this all out to stdout?